contracts/votes/src/balance.rs

```diff
@@ -1,15 +1,119 @@
-use crate::{error::TokenVotesError, storage};
+use crate::{
+    checkpoints::{add_supply_checkpoint, Checkpoint},
+    error::TokenVotesError,
+    storage,
+    voting_units::move_voting_units,
+};
 use soroban_sdk::{panic_with_error, Address, Env};
 
-pub fn receive_balance(e: &Env, address: &Address, amount: i128) {
-    let balance = storage::get_balance(e, address);
-    storage::set_balance(e, address, &(balance + amount));
+#[cfg(feature = "emissions")]
+use crate::emissions;
+
+/// Add tokens to an address's balance and update emissions and voting power
+///
+/// ### Arguments
+/// * `to` - The address to add the balance to
+/// * `amount` - The amount to add. This function does nothing if the amount is not greater than 0.
+///
+/// ### Panics
+/// If the total suppy exceeds the maximum value of the checkpoint
+pub fn mint_balance(e: &Env, to: &Address, amount: i128) {
+    if amount > 0 {
```

Comment:
Is this check needed? You already check the amount in the mint call

Reply:
we likely don't need it, but we call mint balance in other places (like emissions), so having the check here seems reasonable